dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerOwnershipManager.java

```diff
@@ -123,6 +122,12 @@ public void init()
             .expireAfterWrite( TEMPORARY_OWNERSHIP_VALIDITY_IN_HOURS, TimeUnit.HOURS )
             .withMaximumSize( 100000 )
             .build();
+
+        ownerCache = cacheProvider.newCacheBuilder( OrganisationUnit.class )
+            .forRegion( "OrganisationUnitOwner" )
+            .expireAfterWrite( 5, TimeUnit.MINUTES )
+            .withMaximumSize( 100 )

```

Comment:
Maybe 1000 is better? In a bulk import, its natural to have that many often.

Reply:
Sure, that sounds fine!